internal/app/go-filecoin/storage_miner_connector/connector.go

```diff
@@ -366,7 +322,42 @@ func (m *StorageMinerNodeConnector) GetSealSeed(ctx context.Context, preCommitMs
 			return
 		}
 
-		m.newListener <- chainsampler.NewHeightThresholdListener(h+interval, sc, ec, ic, dc)
+		m.chainHeightScheduler.AddListener(h + interval)
+	}()
+
+	// translate tipset key to seal seed handler
+	go func() {
+		for {
+			select {
+			case key := <-hc:
```

Comment:
Blocking: I don't understand how anything can get sent on this channel as it is not passed to any function, any field of m or returned from this function.

Reply:
I suspect this hc should be the newly created listener's hit channel above.